### ISSUE TITLE
ArrayBuffer transfer issue with openpgp worker

### ISSUE THREAD

**Issue by SardineFish** [openpgpjs/openpgpjs#886]
Using openpgpjs with WebWorker to decrypt same binary message twice got error due to the object transfer between the Worker. 
Is it an bug or feature? 
```
openpgp.js:36774 Uncaught (in promise) Error: Error during parsing. This message / key probably does not conform to a valid OpenPGP format. Error: Error during parsing. This message / key probably does not conform to a valid OpenPGP format.
    at Object.read (openpgp.js:36773)
```
Simply try this code:
```javascript
openpgp.initWorker({ path: "./openpgp.worker.js" });
var enc = await openpgp.encrypt({
	message: openpgp.message.fromText("Hello World!"),
	armor: false,
	publicKeys: [pubkey],
});
var bin = enc.message.packets.write();
// -> Uint8Array(341) [ ... ]

await openpgp.decrypt({
	message: await openpgp.message.read(bin),
    privateKeys: [prvkey],
});
// -> {signatures: Array(0), data: "Hello World!", filename: "msg.txt"}

bin
// -> Uint8Array [<unreadable>]

await openpgp.decrypt({
	message: await openpgp.message.read(bin),
    privateKeys: [prvkey],
});
// Error
// openpgp.js:36774 Uncaught (in promise) Error: Error during parsing. This message / key probably does not conform to a valid OpenPGP format. Error: Error during parsing. This message / key probably does not conform to a valid OpenPGP format.
//     at Object.read (openpgp.js:36773)
```

**Comment by twiss** [comment]
Hey @SardineFish, thanks for reporting this!

> Is it an bug or feature?

A bit of both :) There's a config option, `openpgp.config.zero_copy = true`, that is normally supposed to enable this, and it's supposed to improve performance. However, it's a bug that this is happening without that config option. The reason this happens is that we also zero-copy the result of `openpgp.message.readArmored`, as the input to that is a string so it's not observable. We also zero-copy the result of `openpgp.message.read`, but that depends on the input, which causes the input to become unreadable, which was an oversight.

Fixing this might be a bit annoying as we'd have to distinguish between Uint8Arrays coming from `readArmored` (which we can always safely zero-copy) and ones coming from `read`. If the performance difference isn't too big, maybe we should just not zero-copy by default for `readArmored` either. Do you have an opinion on how important this is to fix?

Also, you may already know this but as a workaround you can do
```js
await openpgp.message.read(bin.slice())
```

**Comment by SardineFish** [comment]
Thanks for reply!
I have made a binary copy in my project to escape this issue. 
And I think it would be better to add a notice in guide of the openpgp.js. This issue, in my project , seems like that I got a different result with completely same input (I think, before I found the problem), which confuse me for a while. _(:з)∠)_

**Comment by twiss** [comment]
This should be fixed in #894 :)
